packages/odyssey-react/src/components/Text/index.tsx

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type {
+  FunctionComponent,
+  ReactText,
+  ReactNode,
+  HTMLAttributes,
+} from "react";
+import { useCx, useOmit } from "../../utils";
+import styles from "./Text.module.scss";
+import ScreenReaderText from "../ScreenReaderText";
+
+export interface Props extends HTMLAttributes<HTMLElement> {
+  /**
+   * Text content to be rendered
+   */
+  children: ReactText | ReactNode;
+
+  /**
+   * Prepends visually hidden assistive text to the child contents
+   */
+  a11yStart?: ReactText;
+
+  /**
+   * Appends visually hidden assistive text to the child contents
+   */
+  a11yEnd?: ReactText;
+
+  /**
+   * The semantic element to be rendered in to the DOM
+   * @default span
+   */
+  as?:
+    | "address"
+    | "span"
+    | "dfn"
+    | "p"
+    | "abbr"
+    | "em"
+    | "strong"
+    | "sup"
+    | "sub"
+    | "blockquote"
+    | "cite"
+    | "del"
+    | "pre"
+    | "var"
+    | "q"
+    | "s"
+    | "samp"
+    | "small"
+    | "kbd"
+    | "ins"
+    | "mark"
+    | "code";
+
+  /**
+   * The text color style for the text content.
+   * @default body
+   */
+  color?: "body" | "bodyInverse" | "code" | "danger" | "dangerDisabled" | "sub";
+
+  /**
+   * The font weight for the text content.
+   * @default regular
+   */
+  weight?: "regular" | "bold";
+
+  /**
+   * The font style (normal or italic) for the text content.
+   * @default normal
+   */
+  fontStyle?: "normal" | "italic";
+
+  /**
+   * The text-transform for the text content.
+   * @default none
+   */
+  transform?:
+    | "none"
+    | "capitalize"
+    | "uppercase"
+    | "lowercase"
+    | "fullWidth"
+    | "fullSizeKana";
+
+  /**
+   * The font-size for the text content.
+   * @default normal
+   */
+  size?: "lede" | "base" | "caption";
```

Comment:
these names feel very specific to our particular Odyssey theme. I thought we were going to expose these as x-small, small, medium, large, x-large?

Reply:
Unless we plan on exposing additional sizes as L2 tokens, the approach here seems correct.

For designers, it's ideal to map styling props to available tokens because token naming / availability constitutes something similar to a Design API / Guideline. Matching available tokens in Figma is one example. If someone wants their `$ods-text-size-base` to be larger, that value should be re-themed rather than using an arbitrary size modifier on a distinct piece of copy.

We can always iterate on the standardized sizes as use cases become apparent.

Reply:
This was also my thought, I also expect these values will change as we go through the process of defining and implementing our tokens!